src/log.c

```diff
@@ -62,6 +63,8 @@ static const char * const log_levels[] = {
 	[FI_LOG_MAX] = NULL
 };
 
+static struct fi_provider core_prov = FI_CORE_PROVIDER;
```

Comment:
Have 1 core provider and just share the definition between files.  Don't declare the same variable with the intent that it have exactly the same value twice.